solutions/bronze/usaco-1301.mdx

```diff
@@ -0,0 +1,83 @@
+---
+id: usaco-1301
+source: USACO Bronze 2022 February
+title: Watching Mooloo
+author: Chu Minjae
+---
+[Official Analysis (C++, Java, Python)](http://www.usaco.org/current/data/sol_prob3_bronze_feb23.html)
+
+## Explanation
+
+This question is solved through greedy algorithm.
+
+To start with setting variables, as the input varies from 1 to 10^14, it is better to use ```long long``` rather than ```integer``` types.
+
+Variable ```s``` is used to store the previous index in the source code.
+
+The array is used to store the days that Bessie will watch Mooloo.
+
+Variable ```answer``` will be used to store the minimum amount of money she needs to pay. Using variables ```n``` and ```k```, we can get the input.
+
+Next using the ```for loop```, we can get the specific date that Bessie will watch Mooloo.
+
+To be specific, the for loop gets the input starting from index 1 to reduce the possibility of confusion since this problem is closely related to the date.
+
+Next, we start with putting ```k +1``` into the ```answer``` variable because when ```n``` equals 1,the subscription fee will be ```k + 1``` as she still needs to subscribe it for one day even if the ```for loop``` does not operate, and it is easier to calculate the date as we need to add 1 after subtracting two dates to find out the number of dates passed.
+
+We start the ```for loop``` from 2 to find the value of the array at the current index and previous index.
+
+Now, I will talk about the logic. To achieve the minimum amount of money needed to pay, she needs to pay for each day if the cost of subscribing to watch Mooloo for consecutive days is bigger than paying each day.
+
+On the other hand, she needs to pay for consecutive days if the cost of subscribing for each day is more expensive than paying for consecutive days.
+
+This logic is shown through the ```if-else``` statement.
+
+If the consecutive days of two consecutive indexes of the array are bigger than k, it is better to subscribe for one day each.
+
+If it is not, it is better to subscribe for consecutive days. I will start with the if part. If ```arr[i] - arr[s]``` which is the number of consecutive days is bigger than k, it is better to subscribe each day for one day which is shown as ```ans+= k + 1```.
+
+On the other hand, if it is not, it is better to subscribe for consecutive days which is shown as ```ans+= arr[i] - arr[s]```.
+
+There is no need to add 1 because 1 is already added at the initial value of the answer variable. By repeating this logic through the for loop, we can get the minimum amount of money she needs to pay.
+
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+long long arr[100000];  // long long type should be used as the input can exceed
```

Comment:
unnecessary, make this a vector, set size to n, and move inside main function

Reply:
Changed it